src/interactions/levels/Leaderboard.ts

```diff
@@ -19,29 +19,32 @@ export default class GiveXP extends SubCommand {
   ) {
     await interaction.deferReply();
 
-    const data = (await UserModel.find({ guild_id: interaction.guildId! }))
-      .sort((a: IUser, b: IUser) => b.xp - a.xp)
-      .splice(0, 10);
+    const data = (await UserModel.find({ guild_id: interaction.guildId!, xp: { $ne: 0 } })).sort(
+      (a: IUser, b: IUser) => b.xp - a.xp,
+    );
 
     const embed = this.bot.utils
       .baseEmbed(interaction)
       .setTitle(`${interaction.guild?.name} ${lang.LEVELS.LEADERBOARD}`);
 
-    await Promise.all(
-      data.map(async (item: IUser, idx: number) => {
-        const userId = item.user_id;
-        const member = await this.bot.utils.findMember(interaction, [userId]);
+    let idx = 0;
+    for (const user of data) {
+      const member = await interaction.guild?.members.fetch(user.user_id).catch(() => null);
+
+      if (member) {
         const isInPlace = [0, 1, 2].includes(idx);
 
-        if (member) {
-          embed.addField(
-            member.user.username,
-            `${isInPlace ? places[idx] : ""} ${this.bot.utils.formatNumber(data[idx].xp)}xp`,
-            true,
-          );
-        }
-      }),
-    );
+        embed.addField(
+          member.user.username,
+          `${isInPlace ? places[idx] : ""} ${this.bot.utils.formatNumber(user.xp)}xp`,
+          true,
+        );
+
+        ++idx;
+      }
+
+      if (idx === data.slice(0, 10).length - 1) break;
```

Comment:
A `for` loop may be even better here? `for (let i = 0; i < 10; i++)`

Reply:
`for (let i = 0; i < 10; i++)` ~ 0.9sec -> https://i.xaliks.xyz/oseJ.png
`for (const user of data)` ~ 0.85sec -> https://i.xaliks.xyz/bnGN.png

Reply:
It's only a few milliseconds, I'd rather have readability over speed here :)